app/decorators/service_provider_session_decorator.rb

```diff
@@ -42,8 +46,8 @@ def sp_name
   end
 
   def sp_return_url
-    if request_url.present?
-      OpenidConnectRedirector.from_request_url(request_url).decline_redirect_uri
+    if sp.redirect_uri.present? && openid_connect_redirector.validate
```

Comment:
I think we want `.valid?` instead of `validate` -- `validate` just fires the checkers, I'm not sure what it returns